pkg/controller/nodelifecycle/node_lifecycle_controller.go

```diff
@@ -1318,12 +1466,28 @@ func (nc *Controller) cancelPodEviction(node *v1.Node) bool {
 	return false
 }
 
-// evictPods queues an eviction for the provided node name, and returns false if the node is already
-// queued for eviction.
-func (nc *Controller) evictPods(node *v1.Node) bool {
+// evictPods:
+// - If node is not marked as evicted, it will be added to the evictor queue.
+//   Returns false if the node name was already enqueued.
+// - If node is marked as evicted, pods are deleted immediately.
```

Comment:
- deletes pods immediately if node is already marked as evicted

Reply:
Done.